packages/back-end/src/models/ExperimentModel.ts

```diff
@@ -107,18 +121,377 @@ const experimentSchema = new mongoose.Schema({
   ideaSource: String,
 });
 
-export const ExperimentModel = mongoose.model<ExperimentDocument>(
+type ExperimentDocument = mongoose.Document & ExperimentInterface;
+
+const ExperimentModel = mongoose.model<ExperimentDocument>(
   "Experiment",
   experimentSchema
 );
 
+/**
+ * Convert the Mongo document to an ExperimentInterface, omitting Mongo default fields __v, _id
+ * @param doc
+ */
 const toInterface = (doc: ExperimentDocument): ExperimentInterface =>
-  omit(doc.toJSON(), ["__v", "_id"]) as ExperimentInterface;
+  omit(doc.toJSON(), ["__v", "_id"]);
 
-type FindOrganizationOptions = {
-  experimentId: string;
-  organizationId: string;
-};
+export async function getExperimentById(
+  organization: string,
+  id: string
+): Promise<ExperimentInterface | null> {
+  const experiment = await ExperimentModel.findOne({ organization, id });
+  return experiment ? toInterface(experiment) : null;
+}
+
+export async function getAllExperiments(
+  organization: string,
+  project?: string
+): Promise<ExperimentInterface[]> {
+  const query: FilterQuery<ExperimentDocument> = {
+    organization,
+  };
+
+  if (project) {
+    query.project = project;
+  }
+
+  return (await ExperimentModel.find(query)).map((m) => toInterface(m));
```

Comment:
This should work too since it has the same signature:

```suggestion
  return (await ExperimentModel.find(query)).map(toInterface);
```

Reply:
Woah! I had no idea this shorthand existed. Very cool.